### DISCUSSION TITLE
DB Migration (Postgres): all tables truncated

### DISCUSSION THREAD

**Discussion by adventurist** [gogs/gogs#6857] [Q&A] [answered]
Greetings!

I've been trying to migrate my gogs database from MySQL to Postgres. I simply used `pgloader` and, though it seems to have been a success, my gogs instance presents no pre-existing data.

After a bit of troubleshooting, it appears that the migration is successful, but all the tables are truncated once I launch gogs. Any idea why this might be?

Version
```
VERSION:
   0.12.3
```

I see that a guide is referenced from a [pre-existing issue](https://github.com/gogs/gogs/issues/1388), but `discuss.gogs.io` appears to be down.

```
Hi @aventrax , forgot to update this issue. This feature has been supported quite a while, you can read more about how to do that: https://discuss.gogs.io/t/how-to-backup-restore-and-migrate/991
```

**Comment by unknwon** [comment, marked as answer]
Hey! There is web archive for this page: https://web.archive.org/web/20180521083825/https://discuss.gogs.io/t/how-to-backup-restore-and-migrate/991

I think it is better to upgrade to 0.12.5 first (in case of bug fixes related to backup and restore).

**Reply by adventurist** [reply]
> Hey! There is web archive for this page: https://web.archive.org/web/20180521083825/https://discuss.gogs.io/t/how-to-backup-restore-and-migrate/991
> 
> I think it is better to upgrade to 0.12.5 first (in case of bug fixes related to backup and restore).

Fantastic! This was very simple and it worked perfectly.

Thank you very much!

**Reply by unknwon** [reply]
FYI I migrated this post to Discussions: https://github.com/gogs/gogs/discussions/6876
